lib/List.dbl

```diff
@@ -36,3 +161,95 @@ pub let iter2 {`re : {type X} -> Unit ->[|_] X} =
     | _                => `re ()
   end
   in iter
+
+pub let init (len : Int) f = 
+  let rec init_aux (i : Int) l f = if i > l then [] else (f i) :: init_aux (i+1) l f 
+  in init_aux 0 (len - 1) f
+
+pub let rec fold_left f acc xs = 
+  match xs with
+  | []      => acc 
+  | x :: xs => fold_left f (f acc x) xs 
+  end 
+
+pub let rec fold_right f xs acc =
+  match xs with 
+  | []      => acc 
+  | x :: xs => f x (fold_right f xs acc)
+  end 
```

Comment:
Should we keep the order of parameters in `fold_left`/`fold_right` the same as in OCaml, or just use convention from Haskell? I prefer OCaml style, but we can discuss it.